### PR TITLE
feat: first version of helm chart

### PR DESCRIPTION
**Pull request by thschue**
## This PR
* Adds the first version of a helm chart
* Adds simple workflow to release the chart

### CODE REVIEW

File: charts/glasskube-operator/templates/serviceAccount.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "glasskube-operator.fullname" . }}
+  namespace: {{ .Release.Namespace }}
```
Comment (kosmoz) [inline charts/glasskube-operator/templates/serviceAccount.yaml:6]: Again, I think the namespace is supplied by helm.

**Review by kosmoz — CHANGES_REQUESTED** [review]
Generally the chart works great! :tada: 

I did find some issues though, which we should address.

Also, the CRDs are currently auto-generated and placed in the deploy directory by a gradle task. We should probably do something similar for the chart, however I think it's okay if that's done in a follow-up ticket.

File: charts/glasskube-operator/templates/deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: glasskube-operator
+  namespace: {{ .Release.Namespace }}
```
[the remark this answers is not in the document]
Comment (thschue) [reply charts/glasskube-operator/templates/deploy.yaml]: removed

File: charts/glasskube-operator/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: "0.4.5" # x-release-please-version
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "0.4.5" # x-release-please-version
+
+dependencies:
```
[the remark this answers is not in the document]
Comment (thschue) [reply charts/glasskube-operator/Chart.yaml:26]: I investigated a bit and this is not possible. When we remove them, they would have to be installed manually. If we don't provide the option to install the dependencies with the chart, we should either document this (manual installation) or ensure that the application works without them.

File: charts/glasskube-operator/templates/namespace.yaml
[the remark this answers is not in the document]
Comment (thschue) [reply charts/glasskube-operator/templates/namespace.yaml:1]: removed

File: charts/glasskube-operator/values.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+glasskubeOperator:
+  image:
+    repository: glasskube/operator
+    tag: "" # defaults to Chart.appVersion if unspecified
+  replicaCount: 1
+  resources:
+    limits:
+      cpu: 1000m
+      memory: 512Mi
```
[the remark this answers is not in the document]
Comment (thschue) [reply charts/glasskube-operator/values.yaml]: updated

File: charts/glasskube-operator/templates/deploy.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "glasskube-operator.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      containers:
+      - env:
```
[the remark this answers is not in the document]
Comment (thschue) [reply charts/glasskube-operator/templates/deploy.yaml:21]: Changed this to the Release Name (I think you used "glasskube-operator" as the release name?)

**Comment by thschue** [comment]
> 

Thank you for your review! I addressed the comments, I'd propose that we do the auto-generation of CRDs in a second issue.

**Review by kosmoz — CHANGES_REQUESTED** [review]
I tried the chart again and it works now. :tada: 

If you could implement the following changes, we would be ready to merge this PR:
- After some discussion, we decided that we would prefer to rename the values identifier `glasskubeOperator` to `operator`.
- Some time has passed since this PR was created (this is our bad…), so it is necessary to add the missing CRDs and update the RBAC manifest.
- We decided to keep the dependencies as part of the chart, but they should be disabled by default. This means that for each dependency there should be a `condition` in the `Chart.yaml` and its default value in `values.yaml` should be set to `false`.
- The `appVersion` can be updated to `0.5.2`.
- The MariaDB Operator version can be updated to `0.15.1`, which the latest version of our operator should be compatible with.

**Review by kosmoz — APPROVED** [review]

**Pull request merged**